src/control/lib/hostlist/hostlist_test.go

```diff
@@ -475,6 +477,31 @@ func TestHostList_Etc(t *testing.T) {
 	}
 }
 
+func TestHostList_Slice(t *testing.T) {
+	for name, tc := range map[string]struct {
+		startList *string
+		expOut    []string
+	}{
+		"empty list": {
+			startList: makeStringRef(""),
+			expOut:    []string{""},
```

Comment:
If it's an empty hostlist, would make more sense to me if the slice was itself empty (instead of containing an empty string).

Reply:
It's a little goofy, I admit, but this is what you get back from when `strings.Split()` is called on an empty `hl.DerangedString()` (see hostlist.go:Hostlist.Slice()). I can take a look at improving that in the future.